### ISSUE TITLE
FullyQualifiedName is not escaped correctly

### ISSUE THREAD

**Issue by ahoischen** [formulahendry/vscode-dotnet-test-explorer#21]
F# allows names quoted with double backticks to contain spaces and other special characters. The CLI can handle this, if those names are quoted, otherwise MSBuild interprets the spaces as separate args.

## Example:
```F#
[<Fact>]
let ``This is a test method`` () = Assert.True(true)
```

### Expected:
```bash
dotnet test --filter "FullyQualifiedName~This is a test method"
#...
#Test Run Succesfull.
```

### Actual:
```bash
dotnet test --filter FullyQualifiedName~This is a test method
#MSBUILD : error MSB1008: Only one project can be specified.
#Switch: a
#
#For switch syntax, type "MSBuild /help"
```

The simplest solution would be to just put quotes around the thing and be done with it. However, quotes are legal characters in such names, so they would have to be escaped. This even allows arbitrary code to be executed, if the user selects that test to be run:

```F#
[<Fact>]
let ``; echo "Hello World!";`` () = Assert.True(true)
```
```Bash
dotnet test --filter FullyQualifiedName~; echo "Hello World!";
# Build started, please wait...
# ...
# No test is available in ...
#
# Hello World!
```

I think simply quoting the argument is good enough since most test names don't contain special characters beyond some spaces and escaping strings *correctly* would be quite some work.

**Comment by Ghostbird** [comment]
Aside from the pull request, I've fixed it in my local install in the JS file by changing `FullyQualifiedName~${test.fullName}` to `FullyQualifiedName~'${test.fullName}'`

**Comment by Ghostbird** [comment]
It turns out that FullyQualifiedName filtering does not work at all for me (#26) so I added a fix for that to #25.
I changed the line above to: `DisplayName='${test.fullName}'`

**Comment by janaka** [comment]
@Ghostbird Is this issue now resolved? can you please test with the latest version?

**Comment by Ghostbird** [comment]
I get a very nice message:
```dotnet sdk >=2.1.2 required to retrieve fully qualified test names. Returning non FQ test names.```
Which means that I'll have to wait a while before I test this, dotnet 2.1.0-rc1 is the latest available on my system.

**Comment by janaka** [comment]
Ah. Yeah for mstest and nunit it runs `dotnet vstest` to get the fqn details. And that command is in the newer version of the SDK. 

Out of interest which OS you on @Ghostbird?

**Comment by JasonRitchie** [comment]
With
Dotnet sdk 2.1.3
Fsharp
Dotnet test explorer 0.3.0

FQNs are not quoted correctly. I cannot run individual tests. In the dotnettestexplorer terminal that it runs, I can rerun the last command but including quotes and the test runs.

**Comment by stefanforsberg** [comment]
Fixed with #90

**Comment by Tarmil** [comment]
This fix has been reverted by 4ba12de5756414725b1b6369671421e378eb74d1, it's failing again.
